src/Tizen.NUI.Components/Style/ControlStyle.cs

```diff
@@ -56,6 +64,63 @@ public ControlStyle(ControlStyle style) : base(style)
             this.CopyFrom(style);
         }
 
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        ~ControlStyle()
+        {
+            if (!isDisposeQueued)
+            {
+                isDisposeQueued = true;
+                DisposeQueue.Instance.Add(this);
+            }
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Dispose()
+        {
+            //Throw excpetion if Dispose() is called in separate thread.
+            if (!Window.IsInstalled())
+            {
+                throw new global::System.InvalidOperationException("This API called from separate thread. This API must be called from MainThread.");
+            }
+
+            if (isDisposeQueued)
+            {
+                Dispose(DisposeTypes.Implicit);
+            }
```

Comment:
How will this code be called?
The code that makes isDisposeQueued true is only in the destructor,
Dispose method cannot be called after Destructor.

Reply:
As I understand,
when the instance is added to the DisposeQueue in destructor,
```
// ControlStyle.cs
~ControlStyle()
{
    if (!isDisposeQueued)
    {
        isDisposeQueued = true;
        DisposeQueue.Instance.Add(this);
    }
}
```
The queue executes callback that calls `dispoable.Dispose()`
```
// DisposeQueue.cs
public void Add(IDisposable disposable)
{
    lock (_listLock)
    {
        _disposables.Add(disposable);
    }

    if (_eventThreadCallback != null)
    {
        _eventThreadCallback.Trigger(); // -> this calls ProcessDisposables() in main thread;
    }
}

public void ProcessDisposables()
{
    lock (_listLock)
    {
        foreach (IDisposable disposable in _disposables)
        {
            disposable.Dispose();
        }
        _disposables.Clear();
    }
}
```